VinylUI.Tests/WinFormsTests.fs

```diff
@@ -410,3 +427,59 @@ let ``getObjConverter for record option type handles nulls`` () =
     let converter = BindPartExtensions.getObjConverter<Book option> ()
     converter.ToSource null |> shouldEqual None
     converter.ToControl None |> shouldEqual null
+
+type ListControls = ListType | ComboType
+
+let listControls = [ ListType; ComboType ]
+
+[<TestCaseSource("listControls")>]
+let ``Model to view correctly updates SelectedIndex to -1`` controlType =
+    use form = new FakeForm()
+    let ctrl =
+        match controlType with
+        | ComboType -> form.ComboBox :> ListControl
+        | ListType -> form.ListBox :> ListControl
+
+    Bind.model(<@ model.Books @>).toDataSource(ctrl, <@ fun b -> b.Id, b.Name @>) |> ignore
+    let getList () = ctrl.DataSource :?> Book seq |> Seq.toList
+    getList () |> shouldEqual model.Books
```

Comment:
what is the purpose of this assertion? `toDataSource` is not the function under test.